astropy/time/tests/test_delta.py

```diff
@@ -685,3 +685,100 @@ def test_no_units_warning():
     # table column with units
     table = Table({"t": [1, 2, 3] * u.s})
     assert np.all(TimeDelta(table["t"]).to_value(u.s) == [1, 2, 3])
+
+
+quantity_str_basic_cases = [
+    # Simple seconds (seconds always have a decimal point)
+    ("1s", "1.0s", 1.0),
+    # Simple minutes
+    ("1min", "1min", 60.0),
+    # Float hours
+    ("2.5hr", "2hr 30min", 2.5 * 3600),
+    # Variations on single input component with exponent to multiple output components
+    ("3.000001e7s", "347d 5hr 20min 10.0s", 30000010.0),
+    ("3.e7s", "347d 5hr 20min", 30000000.0),
+    ("3e7s", "347d 5hr 20min", 30000000.0),
+    # High precision seconds
+    ("1.0123456789012345s", "1.012s", 1.0123456789012345),
+    # High precision seconds, random/missing white space and a longer time interval
+    ("  100.0 d1.0123456789012345 s ", "100d 1.012s", 100 * 86400 + 1.0123456789012345),
+    # All possible components
+    (
+        "2yr 3d 4hr 5min 6.789s",
+        "2yr 3d 4hr 5min 6.789s",
+        2 * 365.25 * 86400 + 3 * 86400 + 4 * 3600 + 5 * 60 + 6.789,
+    ),
+    # Float values in components get normalized
+    (
+        "2.5yr 3.5d 4.5hr 5.5min 6.789s",
+        "2yr 186d 7hr 35min 36.789s",
+        2.5 * 365.25 * 86400 + 3.5 * 86400 + 4.5 * 3600 + 5.5 * 60 + 6.789,
+    ),
+]
```

Comment:
Maybe add a test for high precision? But that needs input of both jd1 and jd2 for the test.